my-react-app/src/redux/login/reducer.js

```diff
@@ -0,0 +1,33 @@
+import { actions } from './actions';
+
+const initialState = {
+  token: localStorage.getItem('token'),
+  auth: !!localStorage.getItem('auth'),
+  emailInvalid: false,
+  passwordInvalid: false
+};
+
+export function reducerLogin(state = initialState, action) {
+  switch (action.type) {
+    case actions.GET_TOKEN:
+      return {
+        state
+      };
+    case actions.GET_TOKEN_SUCCESS:
+      localStorage.setItem('auth', action.payload.auth);
+      localStorage.setItem('token', action.payload.token);
+      return {
+        state,
+        token: action.payload.token,
+        auth: true
+      };
+    case actions.GET_TOKEN_FAILURE:
+      localStorage.setItem('auth', action.payload.auth);
+      localStorage.setItem('token', '');
+      return {
+        state
+      };
+    default:
+      return { state };
```

Comment:
You are englobing the whole state in an object each time this runs. If you want to keep the same state, `return state`